src/tir/schedule/state.cc

```diff
@@ -201,9 +201,8 @@ class BlockInfoCollector : private StmtVisitor {
     bool is_root_block = srefs_.empty();
     // Calculate `BlockInfo::scope`
     Array<StmtSRef> child_block_srefs = std::move(block_frames_.back());
-    BlockInfo& info =
-        self_->block_info.emplace(scope_root, BlockInfo(BlockScope(child_block_srefs)))
-            .first->second;
+    self_->block_info[scope_root] = BlockInfo(BlockScope(child_block_srefs));
+    BlockInfo& info = self_->block_info[scope_root];
```

Comment:
Why this change? I suppose they are logically equivalent and the original one saves one lookup?

Reply:
`emplace` will not overwrite the old entry if the key exists. This function may be called in `UpdateScopeBlockInfo`

Reply:
Oh I see - that makes sense to me! Then to save one look-up, let's do

```suggestion
    BlockInfo& info = self_->block_info[scope_root] = BlockInfo(BlockScope(child_block_srefs));
```